### ISSUE TITLE
 Error in the LMM while migrating a VM

### ISSUE THREAD

**Issue by migvasc** [simgrid/simgrid#345]
# Error in the LMM while migrating a VM

## Scenario:

Using the cluster_multi.xml platform from the examples and considering two homogeneous physical machines (PMs). Those PMs have a speed of 1 gigaflop/s and energy consumption of 1 J/s when using 1 core and 0 J/s when idle.

When I compute 3Gf into a PM without migration, it takes 3 seconds and consumes 3J. 

The VM is created in PM4 and starts computing 3Gf. After 1 second, the VM is migrated to the PM2.

I'm getting the wrong energy consumption when I use the sleep_for in the async execution method. I was using the sleep_for inside a while loop that uses the test() method in order to validate whether the execution has finished.

When I execute showing the debug log, I got an error regarding the LMM for the constraints, and the execution is interrupted:

````
[1.973393] simgrid/src/surf/surf_c_bindings.cpp:66: [surf_kernel/DEBUG] Looking for next event in virtual models
[1.973393] simgrid/src/plugins/vm/VirtualMachineImpl.cpp:146: [surf_vm/DEBUG] assign 0.000000 to vm VM0 @ pm node-0.4cores.org
[1.973393] simgrid/src/kernel/resource/Model.cpp:42: [resource/DEBUG] Before share resources, the size of modified actions set is 0
[1.973393] simgrid/src/kernel/lmm/maxmin.cpp:495: [surf_maxmin/DEBUG] Active constraints : 1
[1.973393] simgrid/src/kernel/lmm/maxmin.cpp:437: [surf_maxmin/DEBUG] MAX-MIN ( '3'(1.000000) '1982'(0.000000) )
[1.973393] simgrid/src/kernel/lmm/maxmin.cpp:440: [surf_maxmin/DEBUG] Constraints
[1.973393] simgrid/src/kernel/lmm/maxmin.cpp:456: [surf_maxmin/DEBUG] 	(1.000000.'3'(1000000000.000000) + 1.000000.'1982'(0.000000) + 0) <= 0.000000 ('69')
[1.973393] simgrid/src/kernel/lmm/maxmin.cpp:459: [root/CRITICAL] Incorrect value (1000000000.000000 is not smaller than 0.000000): 1e+09
````

From what I've seen in the log, this error occurs after the third step of the migration. 

Here are the MWE and the instructions to reproduce it: https://gitlab.com/migvasc/vm_migration_lmm_error

**Comment by agiersch** [comment]
I cannot reproduce the error message with the latest version from branch [master](https://github.com/simgrid/simgrid).
Do you agree?

Here is what I get near the end:
> [...]
> [node-0.4cores.org:mig_wrk:(4) 2.015025] [example/INFO] ######## VM0 migrated: node-0.4cores.org->node-0.2cores.org in 1.01503 s ########
> [VM0:VM0_executor0:(3) 3.000209] [example/INFO] Task executed time: 3.000209
> [node-0.4cores.org:killer:(2) 3.021025] [example/INFO] ALL TASKS FINISHED
> [3.021025] [surf_energy/INFO] Total energy consumption: 2.979393 Joules (used hosts: 2.979393 Joules; unused/idle hosts: 0.000000)
> [3.021025] [example/INFO] Simulation is over
> [3.021025] [surf_energy/INFO] Energy consumption of host node-0.1core.org: 0.000000 Joules
> [3.021025] [surf_energy/INFO] Energy consumption of host node-0.2cores.org: 1.006000 Joules
> [3.021025] [surf_energy/INFO] Energy consumption of host node-0.4cores.org: 1.973393 Joules
> [3.021025] [surf_energy/INFO] Energy consumption of host node-1.1core.org: 0.000000 Joules
> [...]

**Comment by mquinson** [comment]
Hello, I was working on a modified MWE, back in august: https://gitlab.com/mquinson/vm_migration_lmm_error

Even with this, as noted by @agiersch , the code seems to work after applying https://github.com/simgrid/simgrid/commit/91954c5dec0ca32d2b45587f0b6e805d3d6e55d9

@migvasc could you please confirm whether the bug is gone for you or whether something annoying remains?

Thanks, all

**Comment by vascmig** [comment]
Hello,

@mquinson, it was solved after aplying the fix on [91954c5](https://github.com/simgrid/simgrid/commit/91954c5dec0ca32d2b45587f0b6e805d3d6e55d9) .

Thank you very much for the help.

**Comment by migvasc** [comment]
Hello,

@mquinson it was solved with the fix on [91954c5](https://github.com/simgrid/simgrid/commit/91954c5dec0ca32d2b45587f0b6e805d3d6e55d9).

Thank you very much for the help.
